sympy/integrals/integrals.py

```diff
@@ -1404,6 +1406,79 @@ def integrate(*args, **kwargs):
     else:
         return integral
 
+def cauchy_principal_value(func, sym, lower_limit = -oo, upper_limit = oo):
+    """
+    Compute the Cauchy Principle Value of the definite integral of a real function on the entire real axis.
```

Comment:
Not on the entire real axis.

Reply:
I'll correct this @jksuom . Thanks for raising this point.